File: Tutoriels/PHP/Silex/root/php/class/GProcess.php
<?php 
    abstract class GProcess {
        //===============================================
        abstract public function run();
        //===============================================
        public static function Instance() {
            $lType = GConfig::Instance()->getData("PROCESS_TYPE");
            if($lType == "MESSEAGE") return GProcessMessage::Instance();
            if($lType == "ROUTER") return GProcessRouter::Instance();
            return GProcessMessage::Instance(); 
        }
        //===============================================
    }
?>        

File: Tutoriels/Cpp/Qt/z2_Qt_Examples/src/manager/GProcess.cpp
//===============================================
#include "GProcess.h"
#include "GProcessSplitterCustom.h"
#include "GProcessButtonCustom.h"
#include "GConfig.h"
//===============================================
GProcess::GProcess() {

}
//===============================================
GProcess::~GProcess() {

}
//===============================================
GProcess* GProcess::Instance() {
    string lProcess = GConfig::Instance()->getData("PROCESS");
    if(lProcess == "SPLITTER_CUSTOM") return GProcessSplitterCustom::Instance();
    if(lProcess == "BUTTON_CUSTOM") return GProcessButtonCustom::Instance();
    return GProcessSplitterCustom::Instance();
}
//===============================================
